react-native-bare-kit.podspec

```
@@ -28,6 +28,10 @@ Pod::Spec.new do |s|
 
   s.vendored_frameworks = "ios/*.xcframework", "ios/addons/*.xcframework"
 
+  s.header_dir = "BareKit"
+
+  s.module_name = "react_native_bare_kit"
```

```suggestion
  s.module_name = s.name
```

This should always match the specification name.

---

Ah, that won't work as `s.name` comes from package.json and is dashed, and `s.module_name` must be a valid C99 identifier, thus underscores

---

Unless there's a standard way of converting from `react-native-bare-kit` to `react_native_bare_kit`, I'm not very familiar with ruby (podspec is written in ruby I think)

---

Ah, terribly good point! Per the Podspec reference, the specification name is used as the default, but they don't mention that they run it though this method first: https://github.com/CocoaPods/CocoaPods/blob/59465a40b7ec1a23a7edc78b9c3be964bce81605/lib/cocoapods/target.rb#L370-L372

---

It available on the `Target` class, so I imagine we could call it directly?

---

Just tried calling `s.c99ext_identifier(s.name)` and unfortunately it's marked as private so we can't use it directly

---

We could reuse the regex substitution tho, like `s.name.gsub(/^([0-9])/, '_\1').gsub(/[^a-zA-Z0-9_]/, '_')` if that's fine with you

---

Eh, let's leave it at it's current hardcoded value then 👍